--- jquery.wymeditor.js	2010-02-03 00:40:43.000000000 -0500
+++ jquery.wymeditor.flash.js	2010-02-03 00:45:20.000000000 -0500
@@ -2167,8 +2167,43 @@
       },
       "required":[
       "name"
+      ],
+	"inside":"object"
+    },
+"embed":
+    {
+      "attributes":
+      [
+        "width",
+        "height",
+        "allowfullscreen",
+        "allowscriptaccess",
+        "wmode",
+	"flashvars",
+	"pluginspage",
+        "type",
+        "src"
+      ],
+          "inside":"object"
+    },
+"object":
+    {
+      "attributes":[
+      "archive",
+      "classid",
+      "codebase",
+      "codetype",
+      "data",
+      "declare",
+      "height",
+      "name",
+      "standby",
+      "type",
+      "usemap",
+      "width"
       ]
     },
+		
     "30":"pre",
     "q":
     {
@@ -3371,13 +3406,13 @@
     "dfn", "dl", "dt", "em", "fieldset", "form", "head", "h1", "h2",
     "h3", "h4", "h5", "h6", "html", "i", "ins",
     "kbd", "label", "legend", "li", "map", "noscript",
-    "object", "ol", "optgroup", "option", "p", "param", "pre", "q",
+    "object", "ol", "optgroup", "option", "p", "pre", "q",
     "samp", "script", "select", "small", "span", "strong", "style",
     "sub", "sup", "table", "tbody", "td", "textarea", "tfoot", "th",
     "thead", "title", "tr", "tt", "ul", "var", "extends"];
 
 
-    this.inline_tags = ["br", "hr", "img", "input"];
+    this.inline_tags = ["br", "hr", "img", "input", "embed", "param"];
 
     return this;
 };
@@ -4089,8 +4124,9 @@
     
     //replace em by i and strong by bold
     //(designMode issue)
-    html = html.replace(/<em([^>]*)>/gi, "<i$1>")
-      .replace(/<\/em>/gi, "</i>")
+    // Also don't replace <embed> tags
+    html = html.replace(/<em[^bed]([^>]*)>/gi, "<i$1>")
+      .replace(/<\/em[^bed]>/gi, "</i>")
       .replace(/<strong([^>]*)>/gi, "<b$1>")
       .replace(/<\/strong>/gi, "</b>");
     
